components/infra-proxy-service/storage/postgres/postgres.go

```diff
@@ -40,6 +40,21 @@ func New(logger logger.Logger, migrationConfig migration.Config, authzClient aut
 	return &postgres{db, logger, authzClient}, nil
 }
 
+// New instantiates and returns a postgres migration implementation
+func NewMigration(logger logger.Logger, migrationConfig migration.Config, authzClient authz.AuthorizationServiceClient) (storage.MigrationStorage, error) {
+
+	db, err := initPostgresDB(migrationConfig.PGURL.String())
```

Comment:
No need to create another connection pool. May call this function from New() and use the same db object

Reply:
Did the changes, please check